Plugins/NativeEngine/Source/BgfxCallback.h

```diff
@@ -26,5 +28,8 @@ namespace Babylon
         void captureEnd() override;
         void captureFrame(const void* _data, uint32_t _size) override;
         void trace(const char* _filePath, uint16_t _line, const char* _format, ...);
+
+        std::mutex m_ssCallbackAccess;
+        std::vector<Napi::FunctionReference> m_screenshotCallbacks;
```

Comment:
Consider using arcana::blocking_concurrent_queue

Reply:
I tried using arcana::blocking_concurrent_queue but to be able to get an env for creating the uint8array, I had to pop from the queue. Once poped, I can't send the FunctionReference to the lambda because of unmovable members in std::function. I didn't want to create a temporary array that I would pass to the lambda and then do a uint8array in the dispatched lambda. So, I used a std::queue with a mutex.

Reply:
You can use Napi::ArrayBuffer::New with an external buffer instead. That way you can create the buffer using normal C++ and then only create the Napi types inside the dispatch.

Reply:
I have to unswizzle the buffer and the data pointer that bgfx gives me is unmapped outside the screenshot function.